### ISSUE TITLE
optimizer suggestions

### ISSUE THREAD

**Issue by wangleiphy** [DerFahrgast/jax-flow-matching#6]
- [x]  https://github.com/DerFahrgast/jax-flow-matching/blob/haiku/main.py#L5

switch to [optax](https://github.com/deepmind/optax) 

- [x]  https://github.com/DerFahrgast/jax-flow-matching/blob/haiku/main.py#L35

`step` -> `lr`, and tries a smaller learning rate than your default (1e-2), investigate on learning rate decay. 

- [x] More serious stopping strategies: split dateset into training and validation set. Perform Early stop when their losses split. (But if you resample gaussian x0 in the training loop, this could be done differently, it will be more like online learning).

- [x] `jit` the training step
